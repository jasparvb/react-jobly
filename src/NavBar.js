import React, { useContext } from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import UserContext from "./UserContext";

function NavBar({ logout }) {
  const { user } = useContext(UserContext);

  return (
    <div>
      <Navbar className="NavBar" expand="md">
        <NavLink exact to="/" className="navbar-brand">
          Jobly
        </NavLink>

        { user ? 
        <Nav className="ml-auto" navbar>
          <NavItem>
            <NavLink to="/companies">Companies</NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/jobs">Jobs</NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/profile">Profile</NavLink>
          </NavItem>
          <NavItem>
            <NavLink to="/" onClick={logout}>Logout</NavLink>
          </NavItem>
        </Nav>
        : 
        <Nav className="ml-auto" navbar>
          <NavItem>
            <NavLink to="/login">Login</NavLink>
          </NavItem>
        </Nav>
        }
      </Navbar>
    </div>
  );
}

export default NavBar;
