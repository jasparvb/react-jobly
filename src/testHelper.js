import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "test",
  first_name: "test",
  last_name: "mctest",
  email: "test@test.net",
  photo_url: null,
  jobs: []
};

const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
