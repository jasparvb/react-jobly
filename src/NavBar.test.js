import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import { UserProvider } from "./testHelper";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <NavBar />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
