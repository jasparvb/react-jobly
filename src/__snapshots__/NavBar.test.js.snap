// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="NavBar navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/companies"
          >
            Companies
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/jobs"
          >
            Jobs
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="NavBar navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/companies"
          >
            Companies
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/jobs"
          >
            Jobs
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/profile"
          >
            Profile
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
